Expensify/App
# Add ability to dismiss attachment modal via swipe

@marcaaron will you please review this?

### Details
This PR adds the ability to swipe down and right to dismiss an attachment on mobile.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/154788

### Tests
1. Sign into an account and send an image attachment
2. After the attachment uploads, tap on it, confirm an attachment modal shows
3. Double tap on the image, confirm it zooms
4. Double-tap again, confirm it zooms out
5. Use two fingers to zoom in, confirm the image zooms in
6. While zoomed in, swipe down and right and confirm the modal doesn't close, and that the viewport pans around the zoomed image
7. Double-tap again then swipe down or right, confirm the modal dismisses
8. Confirm that there are no regressions on web
    1. Confirm attachments still show up on web
    2. Confirm the modal attachment closes normally on web

### QA Steps
Same as above

### Tested On
